pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -0,0 +1,267 @@
+package describe
+
+import (
+	"fmt"
+	"strings"
+
+	utilerrors "github.com/GoogleCloudPlatform/kubernetes/pkg/util/errors"
+	dot "github.com/awalterschulze/gographviz"
+	"github.com/golang/glog"
+	"github.com/gonum/graph"
+
+	osgraph "github.com/openshift/origin/pkg/api/graph"
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildedges "github.com/openshift/origin/pkg/build/graph"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	"github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	imageedges "github.com/openshift/origin/pkg/image/graph"
+	imagegraph "github.com/openshift/origin/pkg/image/graph/nodes"
+	"github.com/openshift/origin/pkg/util/parallel"
+)
+
+// ChainDescriber generates extended information about a chain of
+// dependencies of an image stream
+type ChainDescriber struct {
+	client.Interface
+	Output string
+}
+
+// NewChainDescriber returns a new ChainDescriber
+func NewChainDescriber(c client.Interface, out string) *ChainDescriber {
+	return &ChainDescriber{Interface: c, Output: out}
+}
+
+// MakeGraph will create the graph of all build configurations and the image streams
+// they point to via image change triggers in the provided namespace
+func (d *ChainDescriber) MakeGraph(namespace string) (osgraph.Graph, error) {
+	g := osgraph.New()
+
+	loaders := []GraphLoader{
+		&bcLoader{namespace: namespace, lister: d.Interface},
+		&isFromBuildConfigLoader{namespace: namespace, lister: d.Interface, bc: &bcLoader{namespace: namespace, lister: d.Interface}},
+	}
+	loadingFuncs := []func() error{}
+	for _, loader := range loaders {
+		loadingFuncs = append(loadingFuncs, loader.Load)
+	}
+
+	if errs := parallel.Run(loadingFuncs...); len(errs) > 0 {
+		return g, utilerrors.NewAggregate(errs)
+	}
+
+	for _, loader := range loaders {
+		loader.AddToGraph(g)
+	}
+
+	buildedges.AddAllInputOutputEdges(g)
+	imageedges.AddAllImageStreamRefEdges(g)
+
+	return g, nil
+}
+
+func (d *ChainDescriber) Describe(namespace, input string) (string, error) {
+	g, err := d.MakeGraph(namespace)
+	if err != nil {
+		return "", err
+	}
+	name, tag, all, err := parseInput(input)
+	if err != nil {
+		return "", err
+	}
+	if all {
+		// TODO: Dump all the graph
+		return "", nil
+	}
+
+	// Find out if the provided stream exists
+	exists := false
+	var stream graph.Node
+	for _, node := range g.NodesByKind(imagegraph.ImageStreamNodeKind) {
+		if g.GraphDescriber.Name(node) == name {
+			stream = node
+			exists = true
+			break
+		}
+	}
+	if !exists {
+		return "", fmt.Errorf("image stream %q does not exist", name)
+	}
+
+	if d.Output == "dot" {
+		return dotDump(g, stream, tag)
+	}
+	return g.String(), nil
```

Comment:
@deads2k hmm, that seems like a show-stopper. It's not just a matter of "traverse from the ist of interest and find everything that depends on it", we need to somehow output that subgraph. This is simple with a recursive structure and the json marshaler. Unfortunately there is no equivalent for dot so we ended up creating dot semi-manually (semi because the syntax is built behind funcs like AddEdge and AddNode). WIth our graph package we'll have to create json manually. I think this refactor is too early to happen.:(

I think I'll have to rework this. Re the issue you reported, that happens because we look in status.tags. A much easier fix for that, instead of this refactor would be to search in the tags founds in buildConfigs. What do you think?